packages/fxa-payments-server/src/components/PaymentMethodHeader/index.test.tsx

```diff
@@ -0,0 +1,118 @@
+
+    await waitForExpect(() => {
+      expect(queryByTestId('header')).toBeInTheDocument();
+      expect(queryByTestId('header-prefix')).not.toBeInTheDocument();
+    });
+  });
+
+  it('render header with prefix', async () => {
+    const plan = MOCK_PLANS[0];
+    const props = { plan, onClick: () => {}, prefix: '2.' };
+    const { queryByTestId } = render(<PaymentMethodHeader {...props} />);
+
+    await waitForExpect(() => {
+      expect(queryByTestId('header')).not.toBeInTheDocument();
+      expect(queryByTestId('header-prefix')).toBeInTheDocument();
+    });
+  });
+
+  describe('Fluent localized text', () => {
+    const bundle = setupFluentLocalizationTest('en-US');
+
+    it('returns correct heading without prefix', async () => {
+      const msgId = 'payment-method-header';
+      const expected = 'Choose your payment method';
+      const actual = getLocalizedMessage(bundle, msgId, {});
+
+      expect(actual).toEqual(expected);
+    });
+
+    it('returns correct heading with prefix', async () => {
+      const msgId = 'payment-method-header-prefix';
+      const prefix = '2.';
+      const expected = '2. Choose your payment method';
+      const actual = getLocalizedMessage(bundle, msgId, { prefix });
+
+      expect(actual).toEqual(expected);
+    });
+
+    it('returns correct text for required text', async () => {
+      const msgId = 'payment-method-required';
+      const expected = 'Required';
+      const actual = getLocalizedMessage(bundle, msgId, {});
+      expect(actual).toEqual(expected);
+    });
+  });
+
+  describe('PaymentConsentCheckbox component - Spot checks', () => {
+    it('Checkbox renders as expected', async () => {
+      const plan = MOCK_PLANS[0];
+      const props = { plan, onClick: () => {} };
+      const { findByTestId } = render(<PaymentMethodHeader {...props} />);
+      const checkbox = await findByTestId('confirm');
+
+      expect(checkbox).toBeVisible();
+    });
+
+    it('onClick works', async () => {
+      const plan = MOCK_PLANS[0];
+      const onClickSpy = jest.fn();
+      const props = { plan, onClick: onClickSpy };
+      const { findByTestId } = render(<PaymentMethodHeader {...props} />);
+      const checkbox = await findByTestId('confirm');
+
+      await act(async () => {
+        fireEvent.click(checkbox);
+      });
+
+      expect(onClickSpy).toHaveBeenCalled();
+    });
+
+    it('Legal text displayed correctly for given plan', async () => {
+      const plan =
+        MOCK_PLANS.find((p) => p.plan_id === 'plan_daily') || MOCK_PLANS[0];
+      const props = { plan, onClick: () => {} };
+      const expectedMsg =
+        'I authorize Mozilla, maker of Firefox products, to charge my payment method <strong>$5.00 daily</strong>, according to <termsOfServiceLink>Terms of Service</termsOfServiceLink> and <privacyNoticeLink>Privacy Notice</privacyNoticeLink>, until I cancel my subscription.';
+
+      const testRenderer = TestRenderer.create(
```

Comment:
Not sure why you switch to `react-test-renderer` here but let's stick to `Testing-Library/react`, which you are already using in the tests above.

Reply:
Cool updated the test and removed the `react-test-render` stuff.